tests/test_download.py

```diff
@@ -1,23 +1,67 @@
+import hashlib
+from io import BytesIO
+
 import pytest
+import requests
+from PIL import Image
 
 from streetview import get_panorama
-from streetview.download import iter_tile_info, iter_tiles
+from streetview.download import (
+    get_width_and_height_from_zoom,
+    iter_tile_info,
+    iter_tiles,
+    make_download_url,
+)
+
+# This MD5 was retrieved empirically by downloading tile with bad coordinates
+BAD_TILE_MD5 = "be32aa9ed3880664433199f9e0615cd6"
+
+
+def get_tile_md5_from_url(pano_id: str, zoom: int, x: int, y: int):
```

Comment:
This function name is a little misleading. It makes a HTTP call. The name suggests that it makes a MD5 hash from the URL, not that it makes a MD5 hash from a HTTP response.

What if you added a `fetch_panorama_tile()` function to `download.py` and then created a `hashtile(tile)` function?

Reply:
Agree, changed it